### PR TITLE
fix(chat): smooth worktree setup status

### PR DESCRIPTION
**Pull request by juliusmarminge**
## What changed

Creating a thread with a new worktree could make the timeline jump and leave a "Received 2 updates" group above the running turn.

Web setup now uses a fixed-height status line: shimmering "Setting up worktree…" changes to "Working for…" when the turn starts. The normal **Thinking** marker then appears below, until assistant content or live tool activity replaces it. Its space stays reserved during setup so the handoff does not shift the timeline. The draft stays mounted through setup and keeps its timeline position when navigation completes.

Routine setup notices are omitted from the web and mobile work logs. Setup errors stay visible. Mobile shows the pending label in its workspace controls. Both shimmer implementations respect reduced motion.

## UI changes

The same completed thread on the base version and this branch:

| Before | After |
| --- | --- |
| Setup notices remain in history | Routine setup notices are removed |

[Successful worktree setup recording](https://gh-file-drop-api-prod-mi5fy3sowv63ufte.pinglabs.workers.dev/f/95995f81bf254119/worktree-success.mp4): a real checkout, the shimmer-to-timer handoff with Thinking, and the completed response. The fixture briefly pauses checkout so the transition is visible.

[Failed worktree setup recording](https://gh-file-drop-api-prod-mi5fy3sowv63ufte.pinglabs.workers.dev/f/84f6f1e1929d05a6/worktree-failure.mp4): a Git checkout hook deliberately exits with an error. Setup feedback clears, the error remains, and the original prompt returns to an editable draft with Send available.

[Ordinary turn before its first response](https://gh-file-drop-api-prod-mi5fy3sowv63ufte.pinglabs.workers.dev/f/a2528147fedd66fc/ordinary-turn.mp4): a current-checkout turn shows the working timer and Thinking immediately, then the response.

<details>
<summary>Failed checkout screenshot</summary>

Checkout error with the original prompt restored

</details>

## Validation

- Focused web and mobile tests cover setup error retention, draft promotion, timeline anchoring, and composer behavior. Two server startup-ordering tests passed.
- The Thinking follow-up passed 49 timeline logic tests and 38 timeline component tests, including empty assistant placeholders, content/tool handoff, and stable-row updates.
- Targeted lint and formatting passed. React Doctor found no new issues in the changed code.
- Web, mobile, and client-runtime typechecks passed.
- Real browser runs verified successful checkout, checkout failure, an ordinary turn before its first response, and history after reload. The status block kept the same position and height through setup, the first Thinking state, and navigation to the thread. The setup shimmer unmounts at handoff.
- Mobile verification covered logic tests and typechecking; the recordings show the web client.

## Checklist

- [x] This PR is small and focused
- [x] I explained what changed and why
- [x] I included before/after screenshots for any UI changes
- [x] I included a video for animation/interaction changes

---

> [!NOTE]
> **Medium Risk**
> Changes draft promotion timing and timeline row derivation, which affect first-send and worktree flows; behavior is covered by new tests but navigation edge cases are user-visible.
> 
> **Overview**
> Improves **worktree setup** so chat stays visually stable: one fixed-height status line shimmers **“Setting up worktree…”** (web timeline and mobile new-task workspace controls), then hands off to **“Working for…”** when the turn starts. The composer footer no longer shows a separate “Preparing worktree…” label.
> 
> **Draft → thread navigation** waits until the server has something meaningful to show: the first turn has `startedAt`, or the session ends in **error / stopped / interrupted**. That keeps the draft mounted through checkout/setup instead of promoting too early.
> 
> **Work logs** on web and mobile drop routine `setup-script.requested` / `setup-script.started` noise via shared `isWorktreeSetupActivity`; **error-toned** setup failures still appear in the feed.
> 
> The active-turn **working row** no longer always shows a placeholder **“Thinking”** line—only when there’s no assistant text and no live tool activity—reducing extra rows at setup-to-turn transitions. Timeline anchoring uses **`CHAT_TIMELINE_ANCHOR_OFFSET` (24px)** so promotion preserves scroll position.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 4b10ed5401355ff9f3e3f0d9bf1c6981b7ba0a64. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

> [!NOTE]
> ### Smooth worktree setup status across mobile and web chat UI
> - Delays draft-to-server thread promotion until the first turn starts or setup ends in an error/stop/interruption, so the draft stays mounted during worktree setup
> - Replaces workspace controls on mobile and working timeline rows on web with a shimmering 'Setting up worktree…' status during setup, hiding the elapsed timer and thinking indicator
> - Filters routine `setup-script.requested` and `setup-script.started` activities from derived work logs unless they are error-toned, via new `isWorktreeSetupActivity` helper in [presentation.ts](https://github.com/pingdotgg/t3code/pull/8922/files#diff-b6852793cd3916fd7bb7704e9bfe1391f7ca44403063a0502eb92d59f5060972)
> - Introduces `ActivityShimmerOverlay` and increases timeline anchor offset from 16 to 24 in [timelineScrollAnchoring.ts](https://github.com/pingdotgg/t3code/pull/8922/files#diff-7d5bb365d2e39e9337a3a31436884ed0c0df1127c2033e30a6c027826c46de67)
> - Risk: `resolveDraftPromotionNavigationTarget` signature changes from boolean to `serverThread` object; callers in [ChatView.logic.ts](https://github.com/pingdotgg/t3code/pull/8922/files#diff-464e876afd6be3800453dd8cc94d805066953bed62ffd825bbd6fdfd86f4d114) and [_chat.draft.$draftId.tsx](https://github.com/pingdotgg/t3code/pull/8922/files#diff-81aad35683c56f7b68b13e851e82c9b2677cce9d19548633e91b9b93798e030a) are updated but out-of-tree consumers would break
>
> 
>
> <sup><a href="https://app.macroscope.com">Macroscope</a> summarized 4b10ed5.</sup>
> 
>

### CODE REVIEW

File: apps/web/src/components/ChatView.logic.ts [hunk 1]
```diff
@@ -110,13 +110,19 @@ export function resolveDraftHeroState(input: {
...
     return null;
   }
-  return input.serverThreadStarted ? input.serverThreadRef : null;
+  const sessionStatus = input.serverThread?.session?.status;
+  const turnStarted = input.serverThread?.latestTurn?.startedAt != null;
+  const startupStopped =
+    sessionStatus === "error" || sessionStatus === "stopped" || sessionStatus === "interrupted";
+  // Keep local preparation feedback mounted until the server can render the
+  // running turn or its startup error on the canonical thread route.
+  return turnStarted || startupStopped ? input.serverThreadRef : null;
```
[the remark this answers is not in the document]
Comment (juliusmarminge) [reply apps/web/src/components/ChatView.logic.ts:125]: The current bootstrap sequence prevents this transition. New threads retain `session: null` while checkout and setup run; [ws.ts awaits both before dispatching the turn](https://github.com/pingdotgg/t3code/blob/3af7cbeb534/apps/server/src/ws.ts#L1077). Provider startup publishes `starting`, and [runtime ready/session-started events preserve that state during a pending turn](https://github.com/pingdotgg/t3code/blob/3af7cbeb534/apps/server/src/orchestration/Layers/ProviderRuntimeIngestion.ts#L1585). The client therefore stays in `connecting`, where [local acknowledgment returns false](https://github.com/pingdotgg/t3code/blob/3af7cbeb534/apps/web/src/components/ChatView.logic.ts#L689).

The two existing server startup-ordering tests pass. The refreshed real-checkout recording retains setup feedback through the handoff. No code change is needed for this report.

File: apps/web/src/components/chat/MessagesTimeline.tsx [hunk 2]
```diff
@@ -1318,26 +1328,37 @@ function ProposedPlanTimelineRow({
...
-        </div>
+    <div className="border-b border-border/60 pb-2 pt-1">
+      <div className="flex h-6 min-w-0 items-baseline px-1 text-sm leading-relaxed text-muted-foreground tabular-nums">
+        <span
+          key={isPreparingWorktree ? "setup" : "working"}
+          className="relative shrink-0 overflow-hidden whitespace-nowrap transition-opacity duration-150 starting:opacity-0 motion-reduce:transition-none"
+        >
+          {isPreparingWorktree ? (
+            <>
+              Setting up worktree…
```
[the remark this answers is not in the document]
Comment (juliusmarminge) [reply apps/web/src/components/chat/MessagesTimeline.tsx]: Removed the separate "Preparing worktree..." footer label in 3af7cbe. Setup feedback now appears only in the timeline status line, with the existing busy and send-button behavior preserved.

The [updated success recording](https://gh-file-drop-api-prod-mi5fy3sowv63ufte.pinglabs.workers.dev/f/95995f81bf254119/worktree-success.mp4) shows the single setup label handing off to the timer and restored Thinking marker. Focused composer/timeline tests and the web typecheck pass.

File: apps/web/src/components/chat/MessagesTimeline.tsx [hunk 3]
```diff
@@ -1318,26 +1328,37 @@ function ProposedPlanTimelineRow({
...
+              <span
+                aria-hidden
+                className="live-activity-focus pointer-events-none absolute inset-y-0 select-none"
+              >
+                <span className="live-activity-focus-counter block">
+                  <span className="live-activity-focus-aligned block text-foreground">
+                    Setting up worktree…
+                  </span>
+                </span>
+              </span>
```
[the remark this answers is not in the document]
Comment (juliusmarminge) [reply apps/web/src/components/chat/MessagesTimeline.tsx]: Fixed in 3af7cbeb534. `WorkingTimelineRow` and `LiveActivityRow` now share `ActivityShimmerOverlay`, which owns the mask/counter structure and hidden overlay markup. The content uses span elements to keep the inline nesting valid. Focused timeline tests, lint, formatting, and the web typecheck pass.

File: apps/web/src/components/chat/MessagesTimeline.logic.ts [hunk 4]
```diff
@@ -996,9 +976,7 @@ function isRowUnchanged(a: MessagesTimelineRow, b: MessagesTimelineRow): boolean
 
   switch (a.kind) {
     case "working":
-      return (
-        a.createdAt === (b as typeof a).createdAt && a.showThinking === (b as typeof a).showThinking
-      );
+      return a.createdAt === (b as typeof a).createdAt;
```
[the remark this answers is not in the document]
Comment (juliusmarminge) [reply apps/web/src/components/chat/MessagesTimeline.logic.ts]: Restored in 4b10ed5. The normal Thinking marker appears for ordinary turns and after worktree setup, until assistant content or active tool work replaces it. It stays hidden during setup, with its space reserved so the handoff does not shift the timeline.

Verified both real flows: [worktree setup → Thinking → response](https://gh-file-drop-api-prod-mi5fy3sowv63ufte.pinglabs.workers.dev/f/95995f81bf254119/worktree-success.mp4) and [ordinary turn](https://gh-file-drop-api-prod-mi5fy3sowv63ufte.pinglabs.workers.dev/f/a2528147fedd66fc/ordinary-turn.mp4). The follow-up passed 49 logic tests and 38 timeline tests, including empty assistant placeholders, content/tool handoff, and stable-row updates.
